optimizely/helpers/validator.py

```diff
@@ -72,6 +72,19 @@ def is_config_manager_valid(config_manager):
   return _has_method(config_manager, 'get_config')
 
 
+def is_event_processor_valid(event_processor):
+  """ Given an event_processor, determine if it is valid or not i.e. provides a process method.
+
+  Args:
+    event_processor: Provides a process method to create user events and then send requests.
+
+  Returns:
+    Boolean depending upon whether event_processor is valid or not.
+  """
+
+  return _has_method(event_processor, 'process')
```

Comment:
Shouldn't this check that it implements `BaseEventProcessor`?

Reply:
Doesn't it provides more flexibility to the implementer. We do the same for is_logger_valid, is_event_dispatcher_valid, is_error_handler_valid, is_config_manager_valid in the same class.

Reply:
Doesn't it provides more flexibility to the implementer. We do the same for is_logger_valid, is_event_dispatcher_valid, is_error_handler_valid, is_config_manager_valid in the same class.

Reply:
Makes sense.